### DISCUSSION TITLE
How to minimize the use of pycuda in docker?

### DISCUSSION THREAD

**Discussion by eric642** [inducer/pycuda#354] [Q&A]
In order to install pycuda, the image will become large. What can I do to reduce the image

**Comment by eric642** [comment]
this is my dockerfile
```
FROM nvidia/cuda:11.4.1-cudnn8-runtime-ubuntu20.04
RUN apt-get install -y python3.8 python3-pip libpython3.8 nvidia-cuda-toolkit && update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.8 150 \
&& pip3 install --no-cache-dir nvidia-pyindex==1.0.9  \
&& pip3 install --no-cache-dir nvidia-tensorrt==8.2.4.2 pycuda==2021.1 \
```

**Comment by eric642** [comment]
After installing pycuda, can I uninstall NVIDIA CUDA toolkit to reduce the image size

**Reply by inducer** [reply]
PyCUDA only needs `libcuda.so`, and PyCUDA itself is tiny. The CUDA toolkit can go, all you need from it is the CUDA header file.

**Reply by eric642** [reply]
okay, thank you
